### ISSUE TITLE
The reserved folder name should only be checked against the root folder

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#17751]
The reserved folder name should only be checked against the root folder and should not be a `name.contains`, it should be `name.equalsIgnoreCase`  

We are seeing issues pushing content that include a specific name:
```
[19/12/19 17:10:24:704 GMT] INFO util.PushPublishLogger: class com.dotcms.enterprise.publishing.remote.handler.FolderHandler : [BundleID: 01DWFJ74CVFQDNEW01AYN6BC25.tar.gz] Folder publish (update) success, ID: 66d546a8-95cf-4736-a011-cda6a4d6d746, Inode: 6bf2b94f-1afc-4c43-b2eb-71149ad7765b, Name: application
[19/12/19 17:10:24:712 GMT] INFO util.PushPublishLogger: class com.dotcms.enterprise.publishing.remote.handler.FolderHandler : [BundleID: 01DWFJ74CVFQDNEW01AYN6BC25.tar.gz] Folder publish (update) success, ID: bbf1c9c6-f089-43b0-ae75-7cadabbd08e3, Inode: cdcaa0f8-07e6-4c9e-a61e-5d0ac3559e77, Name: themes
[19/12/19 17:10:24:717 GMT] ERROR handler.FolderHandler: An error occurred when saving local Folder: com.dotmarketing.portlets.folders.model.Folder@f994fa5[name=dotcms,sortOrder=0,showOnMenu=false,hostId=8a7d5e23-da1e-420a-b4f0-471e7da8ea2d,title=dotcms,filesMasks=,defaultFileType=4b87053a-e40b-48ca-963e-fb0a40ebe614,modDate=2019-07-09 12:32:04.182,iDate=2018-05-30 12:50:41.056,type=folder,owner=,inode=ef6c66ec-ad93-46ab-9c26-915025d5ddf2,identifier=a312913c-ee61-4d11-a805-224a4929b9c3]
[19/12/19 17:10:24:717 GMT] ERROR handler.FolderHandler: -- folderName:/application/themes/dotcms/
[19/12/19 17:10:24:717 GMT] ERROR handler.FolderHandler: -- folderId :a312913c-ee61-4d11-a805-224a4929b9c3
[19/12/19 17:10:24:717 GMT] ERROR handler.FolderHandler: -- host :dotcms.com
[19/12/19 17:10:24:718 GMT] ERROR business.PublisherAPIImpl: Error publishing bundle with ID '01DWFJ74CVFQDNEW01AYN6BC25.tar.gz': Folder can't be saved. You entered a reserved folder name
com.dotcms.publishing.DotPublishingException: Folder can't be saved. You entered a reserved folder name
at com.dotcms.enterprise.publishing.remote.handler.FolderHandler.a(SourceFile:278) ~[ee-5.2.2.jar:?]
at com.dotcms.enterprise.publishing.remote.handler.FolderHandler.handle(SourceFile:82) ~[ee-5.2.2.jar:?]
at com.dotcms.publisher.receiver.BundlePublisher.process(BundlePublisher.java:203) ~[dotcms_5.2.2_834c945.jar:?]
at com.dotcms.rest.PublishThread.run(PublishThread.java:42) ~[dotcms_5.2.2_834c945.jar:?]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
Caused by: com.dotmarketing.portlets.folders.exception.InvalidFolderNameException: Folder can't be saved. You entered a reserved folder name
at com.dotmarketing.portlets.folders.business.FolderAPIImpl.validateFolderName(FolderAPIImpl.java:619) ~[dotcms_5.2.2_834c945.jar:?]
at com.dotmarketing.portlets.folders.business.FolderAPIImpl.save_aroundBody26(FolderAPIImpl.java:582) ~[dotcms_5.2.2_834c945.jar:?]
at com.dotmarketing.portlets.folders.business.FolderAPIImpl$AjcClosure27.run(FolderAPIImpl.java:1) ~[dotcms_5.2.2_834c945.jar:?]
at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_5.2.2_834c945.jar:?]
at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[dotcms_5.2.2_834c945.jar:?]
at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[dotcms_5.2.2_834c945.jar:?]
at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[dotcms_5.2.2_834c945.jar:?]
at com.dotmarketing.portlets.folders.business.FolderAPIImpl.save(FolderAPIImpl.java:579) ~[dotcms_5.2.2_834c945.jar:?]
at com.dotmarketing.portlets.folders.business.FolderAPIImpl.save(FolderAPIImpl.java:625) ~[dotcms_5.2.2_834c945.jar:?]
at com.dotcms.enterprise.publishing.remote.handler.FolderHandler.a(SourceFile:252) ~[ee-5.2.2.jar:?]
... 4 more

```

This limits the impact of the reserved name change.

Acceptance Criteria: 
- [x] Integration test

**Comment by wezell** [comment]
PR: #17752

**Comment by dsilvam** [comment]
PR: https://github.com/dotCMS/core/pull/17762/files

**Comment by dsilvam** [comment]
**Note to QA: After latest changes reserved names can't be used only for root-level folders.** 

Test from both _Site Browser_ and _WebDAV_ the following scenarios: 

| Action     | Site Browser | WebDAV |
|------------|--------------|--------|
| Create New |              |        |
| Copy       |              |        |
| Move       |              |        |
| Rename     |              |        |

List of reserved names:
```
WEB-INF", "META-INF", "assets", "dotcms", "html", "portal","email_backups", 
"DOTLESS", "DOTSASS", "dotAdmin", "custom_elements"

```

**Comment by freddyucv** [comment]
I could create folder with some reserverd word like: "assets", "dotcms", "html", "portal","email_backups".
Really I just got a error with "WEB-INF", "META-INF" (I really don't test all the reserved word)
it was testing directly in the site browser

**Comment by freddyucv** [comment]
Also I could create a 'META_INF' folder with cyberduck

**Comment by dsilvam** [comment]
PR fixing condition: https://github.com/dotCMS/core/pull/17782

**Comment by nollymar** [comment]
This needs works. I am able to create a folder in uppercase under the root folder but if I try to do the same under a subfolder (for example: about-us), the folder name is saved in lowercase

**Comment by dsilvam** [comment]
Not part of the changes: It happens because of https://github.com/dotCMS/core/blob/d1a71361510cb5a0b46d579c94cab923bde2a376/dotCMS/src/main/java/com/dotmarketing/webdav/FolderResourceImpl.java#L62

**Comment by wezell** [comment]
Getting fixed here:

https://github.com/dotCMS/core/issues/17698

**Comment by nollymar** [comment]
Tests results: Success

**Comment by bryanboza** [comment]
This is not merged into the release branch, need to send this back.

**Comment by dsilvam** [comment]
@bryanboza this one was just merged into 5.2.5 and 5.2.6. Sending it back to QA

**Comment by bryanboza** [comment]
After these changes we are able to create pages with reserved words like: "META-INF", "WEB-INF" this at root level, but after create these ones you are unable to access this URL from FE. We are returning a 404 error and we are unable to show the page. 

FYI: Tested on demo, and this is working as expected.

**Comment by dsilvam** [comment]
PR: https://github.com/dotCMS/core/pull/17948

**Comment by dsilvam** [comment]
Note to QA: please retest including cases here  https://github.com/dotCMS/core/issues/17751#issuecomment-580981650

**Comment by nollymar** [comment]
**Passed Internal QA:**
Tests results: OK. I tested the initial issue and the one @bryanboza reported here https://github.com/dotCMS/core/issues/17751#issuecomment-580981650

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2.5 // Postgres // FF
